crates/ruff/src/rules/flake8_bugbear/rules/abstract_base_class.rs

```diff
@@ -123,12 +128,23 @@ pub fn abstract_base_class(
             && is_empty_body(body)
             && !is_overload(&checker.ctx, decorator_list)
         {
-            checker.diagnostics.push(Diagnostic::new(
+            let mut diagnostic = Diagnostic::new(
                 EmptyMethodWithoutAbstractDecorator {
                     name: format!("{name}.{method_name}"),
                 },
                 stmt.range(),
-            ));
+            );
+            if checker.patch(Rule::EmptyMethodWithoutAbstractDecorator) {
+                diagnostic.set_fix(Edit::insertion(
+                    format!(
+                        "@abstractmethod{new_line}{indent}",
+                        new_line = checker.stylist.line_ending().as_str(),
+                        indent = indentation(checker.locator, stmt).unwrap(),
```

Comment:
The end result looks pretty alright, just from eyeballing the snapshots. But not sure the best way to handle this `.unwrap()`, or if there's a better way to match the indentation/newline stuff. Thoughts?

Reply:
I think we should use an `if-let` here and only apply the fix if `indentation` returns `Some`. I'm trying to come up with a scenario in which `indentation` could be `None`... it may not be possible, but it's better to be defensive here.